src/sentry/api/endpoints/organization_discover_saved_queries.py

```diff
@@ -50,6 +51,20 @@ class DiscoverSavedQueriesSerializer(serializers.Serializer):
         allow_null=True,
     )
 
+    def validate_projects(self, attrs, source):
+        organization = self.context['organization']
+        projects = attrs[source]
+
+        org_projects = set(Project.objects.filter(
+            organization=organization,
+            status=ProjectStatus.VISIBLE,
+        ).values_list('id', flat=True))
```

Comment:
Is this ever a big list? You could get a smaller result set by only fetching the requested project ids, and comparing the query result to the request result.

```python
org_projects = set(Project.objects.filter(
    organization=organization,
    id__in=projects,
    status=ProjectStatus.VISIBLE,
).values_list('id', flat=True))

if set(projects) != org_projects:
    raise PermissionDenied
```

Reply:
Yeah, that's better for sure. Updating.